File: backend/app/api/stats.py
"""统计看板API - 数据统计和趋势图表"""

from typing import Optional
from datetime import date
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession
from app.database import get_db
from app.schemas.common import ApiResponse
from app.services.stats_service import get_dashboard_stats, get_daily_stats, get_trend_data
from app.dependencies import get_current_user
from app.models.user import User

router = APIRouter(prefix="/stats", tags=["数据统计"])


@router.get("/dashboard", response_model=ApiResponse)
async def get_dashboard(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """获取看板汇总数据"""
    stats = await get_dashboard_stats(db, current_user.id)
    return ApiResponse(data=stats)


@router.get("/daily", response_model=ApiResponse)
async def get_daily(
    start_date: Optional[date] = Query(None, description="开始日期"),
    end_date: Optional[date] = Query(None, description="结束日期"),
    job_id: Optional[int] = Query(None, description="职位ID筛选"),
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """获取每日统计数据列表"""
    stats = await get_daily_stats(db, current_user.id, start_date=start_date, end_date=end_date, job_id=job_id)
    return ApiResponse(data=stats)


@router.get("/trend", response_model=ApiResponse)
async def get_trend(
    days: int = Query(30, ge=1, le=365, description="统计天数"),
    metric: str = Query("say_hello_times", description="统计指标"),
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """获取趋势数据（用于图表展示）"""
    trend = await get_trend_data(db, current_user.id, days=days, metric=metric)
    return ApiResponse(data=trend)

File: frontend/src/api/stats.ts
/**
 * Webb招聘助手 - 统计数据API
 * 处理仪表盘统计、日报、周报、趋势数据
 */
import { request } from './client';
import type { DashboardStats, DailyProgress } from '../types/models';

/** 获取仪表盘概览统计 */
export const getDashboardStats = () =>
  request.get<DashboardStats>('/stats/dashboard');

/** 获取每日统计数据 */
export const getDailyStats = (date?: string) =>
  request.get<DailyProgress>('/stats/daily', { date: date || '' } as object);

/** 获取每周统计数据 */
export const getWeeklyStats = () =>
  request.get<DailyProgress[]>('/stats/weekly');

/** 获取趋势数据（指定天数） */
export const getTrendData = (days?: number) =>
  request.get<DailyProgress[]>('/stats/trend', {
    days: days || 30,
  } as object);
